src/categories/categories.controller.ts

```diff
@@ -1,44 +1,49 @@
 import { Controller, Get, Post, Body, Patch, Param, Delete } from '@nestjs/common';
 import { ApiOperation, ApiTags } from '@nestjs/swagger';
-import { ZaLaResponse } from 'src/common/helpers/response';
+import { Ok, ZaLaResponse } from '../common/helpers/response';
 import { CategoriesService } from './categories.service';
 import { CreateCategoryDto } from './dto/create-category.dto';
 import { UpdateCategoryDto } from './dto/update-category.dto';
+import { Category } from '../entities/category.entity';
 
 @ApiTags('Categories')
 @Controller('categories')
 export class CategoriesController {
-  constructor(private readonly categoriesService: CategoriesService) {}
+  constructor(private readonly categoryService: CategoriesService) {}
 
-  @Post()
-  create(@Body() createCategoryDto: CreateCategoryDto) {
-    return this.categoriesService.create(createCategoryDto);
+  @Post('/create')
+  @ApiOperation({summary: 'Create a new category'})
+  async create(@Body() createCategoryDto: CreateCategoryDto):Promise<Ok<Category>> {
+    const category = await this.categoryService.createNewCategory(createCategoryDto);
+    return ZaLaResponse.Ok(category, 'Category created', '201')
   }
 
   @Get()
-  findAll() {
-    return this.categoriesService.findAll();
+  @ApiOperation({summary: 'Get all available category'})
+  async findAll(): Promise<Ok<Category[]>>{
+    const allCategories = await this.categoryService.findAllCategory();
+    return ZaLaResponse.Ok(allCategories, 'Ok', '200')
   }
 
   @Get('apis:categoryId')
```

Comment:
correct this path.. "apis/:categoryId" so it can be picked as a request param

Reply:
Resolved ...